pkg/master/master.go

```diff
@@ -778,18 +796,25 @@ func (m *Master) thirdpartyapi(group, kind, version, pluralResource string) *api
 	}
 }
 
-func (m *Master) GetRESTOptionsOrDie(c *Config, resource unversioned.GroupResource) generic.RESTOptions {
-	storageConfig, err := c.StorageFactory.NewConfig(resource)
+type restOptionsFactory struct {
+	deleteCollectionWorkers int
+	enableGarbageCollection bool
+	storageFactory          genericapiserver.StorageFactory
+	storageDecorator        generic.StorageDecorator
+}
+
+func (f restOptionsFactory) GetRESTOptionsOrDie(resource unversioned.GroupResource) generic.RESTOptions {
```

Comment:
> Any reason against sharing the RestOpenFactory between federation and apiserver? Not in genericapiserver of course.

I can't think of one.  I think we'd want a structure that bundles up all the "normal" arguments for API servers that do use etcd, suggests help and flags, and produces the storage for us on the other end.  My initial goal was just to split this out so I could start making the construction run forward and keep pulling bits from `Run`.